aws/eip.go

```diff
@@ -0,0 +1,94 @@
+package aws
+
+import (
+	"time"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+func getOrSetFirstSeenTag(svc *ec2.EC2, address ec2.Address) (time.Time, error) {
+	key := "cloud-nuke-first-seen"
+	layout := "2006-01-02 15:04:05"
+
+	tags := address.Tags
+	for _, tag := range tags {
+		if *tag.Key == key {
+			return time.Parse(layout, *tag.Value)
+		}
+	}
+
+	now := time.Now().UTC()
```

Comment:
Also, if you have a separate `setFirstSeenTag`, it can take a timestamp as a param, which will make it even easier to test that the method does the right thing, as you wouldn't be bound to the system clock.